app/utils/LocationFinder.scala

```diff
@@ -0,0 +1,98 @@
+package utils
+
+import play.api.libs.functional.syntax._
+import play.api.libs.json._
+import play.api.libs.ws._
+import play.api.Play.current
+
+import scala.concurrent.{Await, Future}
+import scala.concurrent.duration._
+import scala.math._
+import scala.util.{Success, Failure}
+
+object LocationFinder {
+  case class Location(lat: Double, lng: Double)
+  case class Place(name: String, location: Location)
+  case class Places(places: Seq[Place])
+
+
+  // Implicit Reads for case classes
+  implicit val locationReads : Reads[Location] = (
+    (JsPath \ "lat").read[Double] and
+    (JsPath \ "lng").read[Double]
+  )(Location.apply _)
+
+  implicit val placeReads : Reads[Place] = (
+    (JsPath \ "name").read[String] and
+    (JsPath \ "geometry" \ "location").read[Location]
+  )(Place.apply _)
+
+  // Explicit Reads for Places case class
+  val placesReads : Reads[Places] = (
+    (JsPath \ "results").read[Seq[Place]].map(Places(_))
+  )
+
+
+  // Concurrent execution context for Future
+  implicit val context = scala.concurrent.ExecutionContext.Implicits.global
+
+
+  // Find Euclidean distance between two GPS coordinates
```

Comment:
Do you think Euclidean distance is sufficient for this? I mean, sure we expect people to be relatively close to each other so it's an approximate plane but still.

Maybe use Haversine formula for something more accurate (even thought the Earth isn't technically a sphere)
http://stackoverflow.com/questions/27928/calculate-distance-between-two-latitude-longitude-points-haversine-formula